server/src/main/java/org/elasticsearch/search/sort/GeoDistanceSortBuilder.java

```diff
@@ -603,7 +603,11 @@ private IndexGeoPointFieldData fieldData(SearchExecutionContext context) {
         MappedFieldType fieldType = context.getFieldType(fieldName);
         if (fieldType == null) {
             if (ignoreUnmapped) {
-                return new LatLonPointIndexFieldData(fieldName, CoreValuesSourceType.GEOPOINT);
+                return new LatLonPointIndexFieldData(
+                    fieldName,
+                    CoreValuesSourceType.GEOPOINT,
+                    (dv, n) -> { throw new UnsupportedOperationException(); }
```

Comment:
Can you add a comment for why this is unsupported?

Reply:
Done.